packages/frontend/src/pages/NameFormPage.js

```diff
@@ -3,29 +3,42 @@ import { useParams, useNavigate } from "react-router-dom";
 import { FullWidthText } from "../components/Input/Text";
 import TextButtonInput from "../components/Input/TextButtonInput";
 import LoadingPage from "./LoadingPage";
+import { toast } from "react-hot-toast";
 
 export default function NameFormPage() {
 	const navigate = useNavigate();
 	const params = useParams();
 
 	const [loading, setLoading] = useState(true);
+
 	useEffect(() => {
-		// TODO: use context instead of localStorage
-		if (localStorage.getItem("chickName")) {
-			navigate(`/flock/${params.coopName}/lobby/`);
-		}
+		async function checkChickAndFlock() {
+			const chickId = localStorage.getItem("chickId");
+			if (chickId) {
+				const chickResponse = await fetch(
+					`${process.env.REACT_APP_API_URL}/flocks/${params.coopName}/chicks/${chickId}`
+				);
+				console.log(chickResponse.status);
+				if (chickResponse.status === 200) {
+					localStorage.setItem("chickName", chickResponse.name);
+					navigate(`/flock/${params.coopName}/lobby/`);
+					return;
+				}
+			}
+
+			localStorage.clear();
 
-		// get the flock info
-		fetch(`${process.env.REACT_APP_API_URL}/flocks/${params.coopName}`)
-			.then((response) => response.json())
-			.then((data) => {
-				// if the flock exists, show user form
+			const flockResponse = await fetch(
+				`${process.env.REACT_APP_API_URL}/flocks/${params.coopName}/basket`
+			);
+			if (flockResponse.status === 200) {
 				setLoading(false);
-			})
-			.catch((error) => {
-				// if the flock doesn't exist, go back to the tutorial page
-				navigate("/");
-			});
+			} else {
+				navigate("/welcome");
```

Comment:
if the flock already exists, shouldn't we navigate back to the lobby instead?

Reply:
The else is for if the flock doesn't exist. If the flock does exist, then the user will be prompted to enter their name as expected.

Reply:
got it